# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/MockDragServiceController.cpp
# Commit: 1fa3e699ce84
# Full Hash: 1fa3e699ce84a2c7562e25e2e3a6c1194372042f
# Author: David Parks <davidp99@gmail.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 2 - Make C++ callers of nsIDragService::GetCurrentSession pass a widget r=gstoll,win-reviewers,geckoview-reviewers,m_kato
#   
#   Content process callers must call nsIDragService::GetCurrentSession with the
#   correct widget under the drag.  For all callers, it is a sanity check.
#   
# ==============================================================================

diff -r 21e7f9ac67f8 -r 1fa3e699ce84 widget/MockDragServiceController.cpp
--- a/widget/MockDragServiceController.cpp	Thu Jul 04 07:48:03 2024 +0000
+++ b/widget/MockDragServiceController.cpp	Thu Jul 04 07:48:03 2024 +0000
@@ -142,10 +142,9 @@
     ds->StartDragSession();
   }
 
-  nsCOMPtr<nsIDragSession> currentDragSession;
-  nsresult rv = ds->GetCurrentSession(getter_AddRefs(currentDragSession));
-  NS_ENSURE_SUCCESS(rv, rv);
+  nsCOMPtr<nsIDragSession> currentDragSession = ds->GetCurrentSession(widget);
 
+  nsresult rv;
   switch (aEventType) {
     case EventType::eMouseDown:
     case EventType::eMouseMove: